lib/connection.js

```diff
@@ -55,6 +65,33 @@ class Connection extends EventEmitter {
     }
   }
 
+  processPacket(packet) {
+    if (packet.id !== this[kNextId]) {
+      this[kPool].set(packet.id, packet);
+      return;
+    }
+    this.emit('packet', packet);
+    ++this[kNextId];
+    this[walkPool]();
+  }
+
+  [walkPool]() {
+    const packet = this[kPool].get(this[kNextId]);
+    if (packet) {
+      this.emit('packet', packet);
```

Comment:
Kind of boilerplate code, same as in `processPacket`, so you can simply write here 
```javascript
this.processPacket(packet)
```